src/openpgp/packet/keymaterial.iced

```diff
@@ -479,7 +515,7 @@ class KeyMaterial extends Packet
         expire_at = generated + expire_in
         if not winner? or (winner.expire_at? and (winner.expire_at < expire_at))
           winner = { expire_at, generated, expire_in }
-      else if (expire_in? and expire_in is 0) or (not expire_in? and not @is_primary())
```

Comment:
Right, that makes sense. I'm trying to make sure I have a good understanding of why we made that split, instead of continuing to use the same function for both, with just the small change with `is_primary`. And I think I understand why, so I'll stop rabbit-holing over email :)